assets/js/googlesitekit/datastore/forms/index.test.js

```diff
@@ -0,0 +1,117 @@
+import { STORE_NAME } from './constants';
+
+describe( 'core/forms store', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	describe( 'actions', () => {
+		describe( 'setValues', () => {
+			it( 'requires the formName param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues();
+				} ).toThrow( 'formName is required for setting values.' );
+			} );
+
+			it( 'requires the formData param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( 'form-name' );
+				} ).toThrow( 'formData must be an object.' );
+			} );
+
+			it( 'requires the formData param to be an object not an array', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( 'form-name', [] );
+				} ).toThrow( 'formData must be an object.' );
+			} );
+
+			it( 'requires the formData param to be an object not a string', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( 'form-name', 'formData' );
+				} ).toThrow( 'formData must be an object.' );
+			} );
+
+			it( 'works with formData where the key value is updated, and the untouched key and updated key return expected values', () => {
+				const formData = { key1: 'value1', key2: 'value2' };
+				const newFormData = { key1: 'value3' };
+
+				registry.dispatch( STORE_NAME ).setValues( 'form-name', formData );
+
+				registry.dispatch( STORE_NAME ).setValues( 'form-name', newFormData );
+
+				const formOriginalValue = registry.select( STORE_NAME ).getValue( 'form-name', 'key2' );
+				expect( formOriginalValue ).toEqual( 'value2' );
+
+				const formUpdatedValue = registry.select( STORE_NAME ).getValue( 'form-name', 'key1' );
+				expect( formUpdatedValue ).toEqual( 'value3' );
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getValue', () => {
+			it( 'works with a formName that does not exist', () => {
+				const formData = { key1: 'value1', key2: 'value2' };
+
+				registry.dispatch( STORE_NAME ).setValues( 'form-name', formData );
+
+				const formValue = registry.select( STORE_NAME ).getValue( 'new-form-name', 'key2' );
+				expect( formValue ).toEqual( undefined );
+			} );
+
+			it( 'works with formData where the key does not exist', () => {
+				const formData = { key1: 'value1', key2: 'value2' };
+
+				registry.dispatch( STORE_NAME ).setValues( 'form-name', formData );
+
+				const formValue = registry.select( STORE_NAME ).getValue( 'form-name', 'key3' );
+				expect( formValue ).toEqual( undefined );
+			} );
+
+			it( 'works with an existing formName', () => {
+				const formData = { key1: 'value1', key2: 'value2' };
+
+				registry.dispatch( STORE_NAME ).setValues( 'form-name', formData );
+
+				const formValue = registry.select( STORE_NAME ).getValue( 'form-name', 'key2' );
+				expect( formValue ).toEqual( 'value2' );
+			} );
+
+			it( 'preserves data from state when new data is assigned', () => {
```

Comment:
This is more of a test of `setValues` than `getValue`, but it's a good test (one of the ones I suggested above). I think this should just be moved into the test for the `setValues` action.